runtime/tools/python/ttrt/common/api.py

```diff
@@ -1089,49 +1089,78 @@ def check_constraints(self):
                 self.tracy_csvexport_tool_path
             ), f"perf tool={self.tracy_csvexport_tool_path} does not exist - rebuild using perf mode"
 
-            ttnn_binary_paths = self.file_manager.find_ttnn_binary_paths(self["binary"])
-            ttmetal_binary_paths = self.file_manager.find_ttmetal_binary_paths(
-                self["binary"]
-            )
-
-            self.logging.debug(f"ttnn_binary_paths={ttnn_binary_paths}")
-            self.logging.debug(f"ttmetal_binary_paths={ttmetal_binary_paths}")
-
-            for path in ttnn_binary_paths:
-                bin = Binary(self.logger, self.file_manager, path)
+            if "binary" not in self:
+                # Load from Capsule instead. only TTNN Path is supported for now
+                bin = Binary(self.logger, self.file_manager, "", self["capsule"])
```

Comment:
where are you assigning this self["capsule"] argument?

Reply:
Capsule is defined from the arguments passed into the `API.Perf` call.

ex:
```
c_args = API.Perf.registered_args
c_args['capsule'] = ttmlir.passes.ttnn_to_flatbuffer(module)

<...>
```

Reply:
gotcha, looks good!